src/ameba.ts

```diff
@@ -27,23 +28,67 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument): void {
+    public execute(document: TextDocument, virtual = false): void {
         if (document.languageId !== 'crystal' || document.isUntitled || document.uri.scheme !== 'file') {
             return;
         }
 
-        const args = [this.config.command, document.fileName, '--format', 'json'];
-        const dir = workspace.getWorkspaceFolder(document.uri)!.uri.fsPath;
-        const configFile = path.join(dir, this.config.configFileName);
+        // When reading from stdin, cannot pass any files via the CLI, otherwise there will be overlaps
+        // and potentially out of date errors
+        const space = workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document);
+        const args = [this.config.command];
+
+        if (!virtual) {
```

Comment:
```suggestion
        const space = workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document);
        const args = [this.config.command];

        // When reading from stdin, cannot pass any files via the CLI, otherwise there will be overlaps
        // and potentially out of date errors
        if (!virtual) {
```